src/Razor/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentMarkupBlockPass.cs

```diff
@@ -327,12 +327,17 @@ public override void VisitHtmlAttribute(HtmlAttributeIntermediateNode node)
                     return;
                 }
 
-                Builder.Append("=\"");
+                // We examine the node.Prefix (e.g. " onfocus='" or " on focus=\"")
+                // to preserve the quote type that is used in the original markup.
+                var quoteType = node.Prefix.Contains("'") ? "'" : "\"";
```

Comment:
Is the `node.Prefix` value always of this form? Does it work to check `node.Prefix.EndsWith` instead of doing a `Contains`?

Reply:
Ah -- good call! `node.Prefix` will always end with the `quoteType` so an `EndsWith` change actually communicates the check a little bit better.